WalletWasabi/WabiSabi/Models/DecompositionAlgs/Decomposition.cs

```diff
@@ -0,0 +1,59 @@
+using NBitcoin;
+using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace WalletWasabi.WabiSabi.Models.DecompositionAlgs
+{
+	public class Decomposition : IEnumerable<Coin>
+	{
+		private List<Coin> Coins { get; } = new();
```

Comment:
As I mentioned above I would use the `NBitcoin.Coin`,  a `Script` or an `ScriptPubkeyType` in order to compute the VSize for inputs and outputs correctly.

Reply:
I'm OK with using the real `Coin`, and also OK with the simpler thing I described [here](https://github.com/zkSNACKs/WalletWasabi/pull/5822#discussion_r634623583) (because it decomposition doesn't need to know the actual script or prevout or anything... we can define an interface for just the amount + size?).

Reply:
Done. Decomposition work with Money[], but Decompose works with NBitcoin.Coin